plugins/inventory/now.py

```diff
@@ -0,0 +1,464 @@
+      - Mutually exclusive with I(groups).
+    type: dict
+    default: {}
+    suboptions:
+      <column>:
+        type: dict
+        description: Column to use when grouping inventory hosts into groups.
+        default: {}
+        suboptions:
+          includes:
+            description:
+              - Create Ansible inventory groups only for records with <column>
+                matching any of the values specified in this list.
+              - For reference fields, you need to provide C(sys_id).
+              - Mutually exclusive with I(excludes).
+            type: list
+            default: None
+          excludes:
+            description:
+              - Create Ansible inventory groups only for records with <column>
+                matching any value except the ones specified in this list.
+              - For reference fields, you need to provide C(sys_id).
+              - Mutually exclusive with I(includes).
+            type: list
+            default: None
+"""
+
+EXAMPLES = r"""
+# A trivial example that creates a host from every record of the
+# ServiceNow cmdb_ci_server table. The ip_address column is used for
+# for ansible host, and server name for inventory hostname.
+# No groups will be created - all the resulting hosts are ungrouped.
+plugin: servicenow.itsm.now
+
+# `ansible-inventory -i inventory.now.yaml --graph` output:
+# @all:
+#  |--@ungrouped:
+#  |  |--DatabaseServer1
+#  |  |--DatabaseServer2
+#  |  |--INSIGHT-NY-03
+#  |  |--MailServerUS
+#  |  |--VMWARE-SD-04
+
+
+# Group hosts automatically, according to values of manufacturer and os columns.
+# Include only records with the specified operating systems.
+# Groups will most likely overlap.
+plugin: servicenow.itsm.now
+group_by:
+  manufacturer:
+  os:
+    includes:
+      - Linux Red Hat
+      - Windows XP
+
+# `ansible-inventory -i inventory.now.yaml --graph` output:
+# @all:
+#  |--@Dell_Inc_:
+#  |  |--DatabaseServer1
+#  |  |--DatabaseServer2
+#  |  |--INSIGHT-NY-03
+#  |--@Lenovo:
+#  |  |--FileServerFloor1
+#  |  |--FileServerFloor2
+#  |--@Linux_Red_Hat:
+#  |  |--DatabaseServer1
+#  |  |--DatabaseServer2
+#  |--@Windows_XP:
+#  |  |--FileServerFloor1
+#  |  |--FileServerFloor2
+#  |  |--INSIGHT-NY-03
+#  |--@ungrouped:
+
+
+# Group hosts into named groups, according to the specified criteria.
+# Below example creates a single group containing hosts that match
+# all the criteria.
+groups:
+  non_windows_prod_servers:
+    classification:
+      includes: [ Production ]
+    os:
+      excludes:
+        - Windows XP
+        - Windows 2000
+        - Windows 2000 Server
+        - Windows 2003 Standard
+
+# `ansible-inventory -i inventory.now.yaml --graph` output:
+# @all:
+#  |--@HP_UX:
+#  |  |--Car-2
+#  |--@Linux_Red_Hat:
+#  |  |--DatabaseServer1
+#  |  |--DatabaseServer2
+#  |  |--lnux100
+#  |  |--lnux101
+#  |--@Solaris:
+#  |  |--MailServerUS
+#  |--@ungrouped:
```

Comment:
Based on the `groups` definition above, I would expect the output to contain a single group called `non_windows_prod_servers` that contains matching records. Is this a copy-paste error maybe?

Reply:
Yes indeed, luckily that's all it was ;) Thank you for spotting this.